### PR TITLE
fix(1906): cherry pick cache folder or file

### PR DESCRIPTION
**Pull request by parthasl**
## Context

Not able to cherry-pick the folder or file in a directory and store or retrieve the cache.

## Objective

This PR will enable build-cache disk-based strategy to cherry-pick the folder or file.

## References

https://github.com/screwdriver-cd/screwdriver/issues/1906

## License

I confirm that this contribution is made under the terms of the license found in the root directory of this repository's source tree and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

**Review by vnugopal — DISMISSED** [review]

File: sdstore/cache2disk.go [hunk 1]
```diff
@@ -63,26 +68,32 @@ func getDirSizeInBytes(path string) int64 {
...
 		_ = writeLog(logger.WARN, logger.FILE, msg)
 	}
 	_ = json.Unmarshal(oldMd5File, &oldMd5)
 
+	if newMd5, err = GenerateMd5(src); err != nil {
+		msg = fmt.Sprintf("not able to generate md5 for directory: %v", src)
+		_ = writeLog(logger.WARN, logger.MD5, msg)
+	}
+
 	if newMd5, err = MD5All(src); err != nil {
```
Comment (jithine) [inline sdstore/cache2disk.go]: this is doing md5 twice ? line 90 also seems to populate `newMd5`

Comment (parthasl) [reply sdstore/cache2disk.go]: nice catch. fixed

File: sdstore/md5helper.go [hunk 2]
```diff
@@ -112,3 +172,54 @@ func MD5All(root string) (map[string]string, error) {
...
+	md5Channel := make(chan md5Hash)
+	defer close(md5Channel)
+
+	files, err := getAllFiles(path)
+	if err != nil {
+		return nil, err
+	}
+	totalFiles := float64(len(files))
+	batchSize := math.Ceil(totalFiles / float64(MaxConcurrencyLimit))
+	fmt.Printf("batch size: %d, concurreny limit: %d, total files: %d\n", int(batchSize), int(MaxConcurrencyLimit), int(totalFiles))
```
Comment (jithine) [inline sdstore/md5helper.go:195]: this could use the `writeLog` fn

**Review by jithine — APPROVED** [review]
small comment on common logger utility, but we can refactor it later.

**Pull request merged**